google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/SubscriberTest.java

```diff
@@ -176,6 +176,29 @@ public void testFailedChannel_fatalError_subscriberFails() throws Exception {
     }
   }
 
+  @Test(expected = IllegalStateException.class)
+  public void testFailedChannel_shutdownBackgroundResources() throws Exception {
+    Subscriber subscriber =
```

Comment:
We shouldn't have to create a new FakeFixedExecutorProvider class.  We can just create our own custom provider that will autoClose.  For example:

```
ExecutorProvider provider =
        new ExecutorProvider() {
          @Override
          public boolean shouldAutoClose() {
            return true;
          }

          @Override
          public ScheduledExecutorService getExecutor() {
            return fakeExecutor;
          }
        };
```

We can then just pass in the provider when we .setExecutorProvider

Reply:
Done! Sorry for the delay.